packages/dds/tree/src/simple-tree/objectNode.ts

```diff
@@ -140,8 +141,20 @@ function createProxyHandler(
 			const fieldInfo = flexKeyMap.get(viewKey);
 
 			if (fieldInfo !== undefined) {
-				const field = getFlexNode(proxy).tryGetField(fieldInfo.storedKey);
-				return field === undefined ? undefined : getProxyForField(field);
+				const flexNode = getFlexNode(proxy);
+				const field = flexNode.tryGetField(fieldInfo.storedKey);
+				if (field !== undefined) {
+					return getProxyForField(field);
+				}
+
+				// Check if the user is trying to read an identifier field of an unhydrated node, which is not supported.
+				if (fieldInfo.schema.kind === FieldKind.Identifier && isMapTreeNode(flexNode)) {
```

Comment:
This check could theoretically be done inside of `MapTreeNode` instead of here, but I opted to do it here because:
1. This code will remain even if `MapTreeNode` undergoes changes, and likewise its corresponding tests are not tied to the `MapTreeNode` tests.
2. `MapTreeNode` (and all `FlexTreeNodes`) have multiple ways of returning fields (e.g. `getBoxed()`, `tryGetField()`, etc.) and there is no single place that we can do this check from within `MapTreeNode`.
3. Catching it here is the closest point of intervention to the user's code, which makes sense because this error is a policy choice meant to guide the user to correctness (as opposed to being a requirement of our system invariants, which it is not).